Merge xen-api-libs-transitional into xapi

psafont opened the pull request:
There's a commit for cleanup after the merge, and one for sorting out the quality gates

robhoes commented:
There are some `ocaml/libs/.git*` files that I think you should remove.

robhoes commented on gzip.opam:
```diff
@@ -0,0 +1,22 @@
+maintainer: "xen-api@lists.xen.org"
+authors: "xen-api@lists.xen.org"
+homepage: "https://xapi-project.github.io/"
+bug-reports: "https://github.com/xapi-project/xen-api-libs-transitional.git"
```
These URLs are now out of date.

psafont commented:
I've cleaned up the directory and I've added the new packages to the gihub workflow now.

BVT and BST look mostly green except for some tests, but it looks that#s because of infrastructure failures, I'm rerunning them
